plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php

```diff
@@ -78,20 +78,18 @@ function handleElement($node) {
 		// Build a cached list of genres by context ID by name
 		if ($genreName) {
 			if (!isset($genresByContextId[$context->getId()])) {
-				$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
+				$genreDao = DAORegistry::getDAO('GenreDAO'); /** @var $genreDao GenreDAO */
 				$genres = $genreDao->getByContextId($context->getId());
 				while ($genre = $genres->next()) {
-					foreach ($genre->getName(null) as $locale => $name) {
-						$genresByContextId[$context->getId()][$name] = $genre;
+					if ($genre->getData('key') == $genreName) {
+						$genreId = $genre->getId();
```

Comment:
Can you add a `break` here? Looks like you got it in the `main` PR.

Reply:
Fixed